appcues/src/main/java/com/appcues/statemachine/StateMachine.kt

```diff
@@ -223,19 +223,17 @@ internal class StateMachine(
         }
     }
 
-    fun stop(dismiss: Boolean) {
-        appcuesCoroutineScope.launch {
```

Comment:
this is the spot of concern - if this launches it's own coroutine, the caller may continue before this `EndExperience` actually completes